ssl/record/rec_layer_s3.c

```diff
@@ -77,11 +77,28 @@ void RECORD_LAYER_release(RECORD_LAYER *rl)
     SSL3_RECORD_release(rl->rrec, SSL_MAX_PIPELINES);
 }
 
+/* Checks if we have unprocessed read ahead data pending */
 int RECORD_LAYER_read_pending(const RECORD_LAYER *rl)
 {
     return SSL3_BUFFER_get_left(&rl->rbuf) != 0;
 }
 
+/* Checks if we have decrypted unread record data pending */
+int RECORD_LAYER_processed_read_pending(const RECORD_LAYER *rl)
+{
+    size_t num_recs, curr_rec;
+    const SSL3_RECORD *rr;
+
+    rr = rl->rrec;
+    num_recs = RECORD_LAYER_get_numrpipes(rl);
+    for (curr_rec = 0;
```

Comment:
You could move the assignments up to the declaration.  Then change the for loop into a while.